Implemented Basic filters

Shruthi-Babu opened the pull request:
Files/folders to check: Context folder, filterReducer.js, Catalogue.js, FilterSection.js, utils/getFilteredData.js

Shruthi-Babu committed:
Basic Router setup done

Shruthi-Babu committed:
getting data from mockbee

Shruthi-Babu committed:
Initial commit

connected to backend. iterating over products and displaying them. created a basic context for the products.

Shruthi-Babu committed:
Implemented filtering

shrushti2000 commented on src/Components/Catalogue.js:
```diff
@@ -1,8 +1,24 @@
+      setData(response);
+      console.log("data: ", data)
+    } )    
+  })
```
useEffect is the place where data fetching happens most of the time. Data fetching means using asynchronous function. . Try using async await here with IIFE. Also , put this block of code inside try and catch so that errors can be handled. 
You can refer this [https://stackoverflow.com/questions/53332321/react-hook-warnings-for-async-function-in-useeffect-useeffect-function-must-ret](https://stackoverflow.com/questions/53332321/react-hook-warnings-for-async-function-in-useeffect-useeffect-function-must-ret)

shrushti2000 commented on src/Components/Catalogue.js:
```diff
@@ -1,28 +1,71 @@
-//   price: "499",
-//   categoryName: "F.R.I.E.N.D.S.",
-//   photo: "/friends1.jpg"
-// }
```
remove this commented code and console.log

shrushti2000 commented on src/Components/Catalogue.js:
```diff
@@ -1,28 +1,71 @@
+    } else {
+        return productData;
+    }
+}
```
sort is an impure function since it mutates the original array. You can do something like 
[...productData].sort((a, b) => b.price - a.price) this will return a new array and not mutate the original array.

shrushti2000 commented on src/context/reducers/filterReducer.js:
```diff
@@ -1,7 +1,22 @@
+const filterReducer = (state, action) => {
+    let {type, payload} = action;
+    console.log(payload);
+    console.log(type);
```
these console.logs can be removed too.

shrushti2000 reviewed (commented):
Left few suggestions . Remove commented code and console.logs .
Overall everything is good. Folder structure is also good. Good job.
Have a look at the deploy  once .It is not showing up the deployed page.

Shruthi-Babu committed:
code refactor and PR1 suggestions

Removed console.logs
Made the useEffect call Async with IIFE
Improved rating filter implementation
Moved some files into pages folder

shrushti2000 reviewed (approved):
Great job @Shruthi-Babu. All changes have been done correctly.PR Looks good to merge.
